pygen/pygen_src/riscv_asm_program_gen.py

```diff
@@ -116,8 +116,8 @@ def gen_program(self):
             self.gen_program_end(hart)
             for hart in range(cfg.num_of_harts):
                 self.gen_data_page_begin(hart)
-                if(cfg.no_data_page):
-                    self.gen_data_page()
+                if not cfg.no_data_page:
```

Comment:
Hi @aneels3 
Please confirm.

Reply:
Hi @hodjat91
I think the commented part should be `self.gen_data_page(hart)`

rest LGTM, thanks!